main.go

```diff
@@ -15,6 +16,10 @@ import (
 	"github.com/gorilla/mux"
 )
 
+const (
```

Comment:
Is this necessary?

Reply:
Functionally not needed, however, it is in play to avoid magic numbers. Although most people will likely be able to read `os.Exit(1)` and know what it means, being explicit might help newer people.

Reply:
Im not convinced, the idea of using codes 2, 1 or 0 is universal (any language if Im not mistaken) on program exit. I would argue for someone new to either look it up or this be caught up in code review; like Dave spotted with one of my pull requests (use of 1 instead of 0 when the program quit successfully)

Reply:
So was thinking a bit more about this, so my stance is that by removing this we introduce a magic number. I have a similar example of where this is done.

HTTP status codes, we typically don't write `200` or `500` even though these are common universal static numbers that have a meaning which most developers will know what they mean. Instead we use `http.StatusOK` and `http.StatusInternalServerError` which are basically just enumerated values.

Reply:
I feel a new starter should learn about exit codes and feel this hides the fact of there being many exit codes. Your example with http enumerated status codes ..... I don't always use them and you will see this in the backend services and api clients when handling response status codes but that is here nor there.

I think I can find a middle ground, a way to keep your explicitness and also keep a useful enumerated list using iota, like so:

```
type ExitCodes int

const
  Success ExitCodes = iota
  Failure
```

See [here](https://blog.learngoprogramming.com/golang-const-type-enums-iota-bc4befd096d3) for a demo using weekdays

Reply:
I've just realised you can't do the above because we would then be using the wrong type